flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -254,16 +254,6 @@ public BufferPool getBufferPool() {
 		return bufferPool;
 	}
 
-	@Override
-	public int getPageSize() {
-		if (bufferPool != null) {
-			return bufferPool.getMemorySegmentSize();
```

Comment:
Looks like `BufferProvider.getMemorySegmentSize` can be also removed.

Reply:
Yes, I would submit a hotfix commit for removing it in this PR.